python/hail/methods/impex.py

```diff
@@ -382,11 +404,49 @@ def import_locus_intervals(path, reference_genome='default', skip_invalid_interv
     :class:`.Table`
         Interval-keyed table.
     """
-    rg = reference_genome._jrep if reference_genome else None
 
-    t = Env.hail().table.Table.importIntervalList(Env.hc()._jhc, path, joption(rg),
-                                                  skip_invalid_intervals)
-    return Table(t)
+    t = import_table(path, comment="@", impute=False, no_header=True,
+                     types={'f0': tstr, 'f1': tint32, 'f2': tint32,
+                            'f3': tstr, 'f4': tstr})
+
+    if t.row.dtype == tstruct(f0=tstr):
+        if reference_genome:
+            t = t.select(interval=hl.parse_locus_interval(t['f0'], reference_genome))
+        else:
+            interval_regex = "([^:]*):(\\d+)\\-(\\d+)"
+
+            def checked_match_interval_expr(match):
+                return hl.cond(hl.len(match) == 3,
+                               locus_interval_expr(match[0], hl.int32(match[1]), hl.int32(match[2]), hl.bool(True), hl.bool(True),
+                                                   reference_genome, skip_invalid_intervals),
+                               hl.null(tinterval(tstruct(contig=tstr, position=tint32))))
+
+            expr = hl.bind(t['f0'].first_match_in(interval_regex),
+                           lambda match: hl.cond(hl.bool(skip_invalid_intervals),
+                                                 checked_match_interval_expr(match),
+                                                 locus_interval_expr(match[0], hl.int32(match[1]),
+                                                                     hl.int32(match[2]), hl.bool(True), hl.bool(True),
+                                                                     reference_genome, skip_invalid_intervals)))
+
+            t = t.select(interval=expr)
+
+    elif t.row.dtype == tstruct(f0=tstr, f1=tint32, f2=tint32):
+        t = t.select(interval=locus_interval_expr(t['f0'], t['f1'], t['f2'], hl.bool(True), hl.bool(True), reference_genome, skip_invalid_intervals))
+
+    elif t.row.dtype == tstruct(f0=tstr, f1=tint32, f2=tint32, f3=tstr, f4=tstr):
+        t = t.select(interval=locus_interval_expr(t['f0'], t['f1'], t['f2'], hl.bool(True), hl.bool(True), reference_genome, skip_invalid_intervals),
+                     target=t['f4'])
+
+    else:
+        raise FatalError("""invalid interval format.  Acceptable formats:
+              `chr:start-end'
```

Comment:
use a single quote instead of a backtick here, we don't use the backtick style anywhere else in Python.